src/components/FeatureViews/Dashboard/Dashboard.js

```diff
@@ -38,21 +39,59 @@ const useStyles = makeStyles((theme) => ({
 
 const Dashboard = () => {
     const classes = useStyles();
-    const user = useSelector(({auth}) => auth) || [];
+    const {email} = useSelector(({auth}) => auth) || [];
+    const emailQuery = { "email": email }
+    const USER_QUERY = gql`query userQuery($email: String="") {
+        accountSearch(query: $email) {
+          results {
+            ... on AdminType {
+              user {
+                email
+                firstName
+                id
+              }
+            }
+          }
+        }
+      }
+      
+      
+      
+      `;
+      
+    const user = useQuery(USER_QUERY, { 
+        variables: emailQuery, 
+    })
+
+    const userFirstName = user?.data?.accountSearch?.results[0]?.user?.firstName;
```

Comment:
This isn't good - as mentioned, you should wait until data is loaded, then you don't need to do any of this. then you can use data directly.